src/main/java/fi/aalto/cs/apluscourses/presentation/CourseSelectionViewModel.java

```diff
@@ -6,20 +6,29 @@
 import java.net.MalformedURLException;
 import java.net.URL;
 import java.util.List;
+import java.util.Optional;
+import org.jetbrains.annotations.NotNull;
 
 public class CourseSelectionViewModel {
 
-  private List<CourseItemViewModel> courses;
+  @NotNull
+  public final ObservableProperty<CourseItemViewModel[]> courses
+      = new ObservableReadWriteProperty<>(new CourseItemViewModel[0]);
+
+  public final ObservableProperty<CourseItemViewModel> selectedCourse
+      = new ObservableReadWriteProperty<>(null);
 
   public final ObservableProperty<String> selectedCourseUrl
       = new ObservableReadWriteProperty<>("", CourseSelectionViewModel::validateCourseUrl);
 
+  /**
+   * A constructor.
+   */
   public CourseSelectionViewModel(List<CourseItemViewModel> courses) {
-    this.courses = courses;
-  }
-
-  public CourseItemViewModel[] getCourses() {
-    return courses.toArray(new CourseItemViewModel[0]);
+    this.courses.set(courses.toArray(CourseItemViewModel[]::new));
+    selectedCourse.addValueObserver(this, (self, course) ->
+        selectedCourseUrl.set(Optional.ofNullable(course)
```

Comment:
Nit: It doesn't really matter here, but I'd like it to be `self.selectedCourseUrl.set(...)` to avoid closure being formed.